packages/block-library/src/button/editor.scss

```diff
@@ -14,15 +16,32 @@
 	margin-bottom: 0;
 	position: relative;
 
-	.editor-rich-text__tinymce {
+	.editor-rich-text__tinymce.mce-content-body {
 		cursor: text;
-		line-height: 1;
+		line-height: $blocks-button__line-height;
 	}
 
-	&:not(.has-text-color) .editor-rich-text__tinymce[data-is-placeholder-visible="true"] + .editor-rich-text__tinymce {
+	// Make placeholder text white unless custom colors or outline versions are chosen.
+	&:not(.has-text-color):not(.is-style-outline) .editor-rich-text__tinymce[data-is-placeholder-visible="true"] + .editor-rich-text__tinymce {
 		color: $white;
+	}
+
+	// Increase placeholder opacity to meet contrast ratios.
+	.editor-rich-text__tinymce[data-is-placeholder-visible="true"] + .editor-rich-text__tinymce {
 		opacity: 0.8;
 	}
+
+	// Don't let the placeholder text wrap in the variation preview.
+	.editor-block-styles__item-preview & {
```

Comment:
I think this should be `.editor-block-preview__content` because it the preview component can be used in several places.